### ISSUE TITLE
Concurrency flag potentially not working

### ISSUE THREAD

**Issue by richardwillars** [vercel/turborepo#339]
### What version of Turborepo are you using?

1.0.13

### Describe the Bug

When using this command on a yarn workspace with about 20 packages
`yarn turbo run deployEnv1 --concurrency=1`

It seems to spin up lots of node processes within a few seconds and my computer grinds to a halt.
![Screenshot 2021-12-17 at 10 30 12](https://user-images.githubusercontent.com/291004/146531026-ad804531-23c9-4414-8e95-c059e72ba1fc.png)
Killing the turbo command removes all the node processes

I have two theories:
- The concurrency flag isn't working and it's processing lots of tasks at the same time
- Maybe it is processing tasks one after another, but it's not killing the node processes when they're finished. If this happened for both cached and non-cached tasks it'd explain why so many node processes get created very quickly, as tasks that are already cached could be spinning up a node process for each task and not destroying on acknowledgement that it's cached (I hope that makes sense)

I appreciate that's quite a vague bug report.. I'm happy to provide logs etc, just let me know what you need / what commands to run in order to debug / replicate.

### Expected Behavior

1 node process up and running at a time

### To Reproduce

`yarn turbo run deployEnv1 --concurrency=1`

**Comment by jaredpalmer** [comment]
What OS are you using?

Can you run with `--profile` flag and upload the json file to Chrome or Edge DevTools Performance tab? Runniny with `--concurrency=1` should be a single line like this:

![CleanShot 2021-12-17 at 08 10 48@2x](https://user-images.githubusercontent.com/4060187/146549574-459e6968-ed23-451e-84be-a00f92d20878.png)

**Comment by richardwillars** [comment]
I took your advice and ran it with the profile flag, and I can confirm it's actually using one thread. I did some more investigation and it turns out Jest was using about 15 threads for each package, and sometimes not killing them after finishing, which resulted in the thread list increasing. Ran jest with --maxWorkers=1 and that seemed to solve that issue as well. 

So fortunately not a bug with this library!

This library is a great concept and execution btw, moving all our stack over to this as fast as possible as it fixes so many issues!
